### PR TITLE
Rvpyth 18 delete/hide restaurant

### PR DESCRIPTION
**Pull request by krizik**
with "add/edit restaurant" fixes

### CODE REVIEW

**Review by OlhaLesk — APPROVED** [review]

**Review by yurapa — APPROVED** [review]

File: administrator/admin.py [hunk 1]
```diff
@@ -3,4 +3,26 @@
...
+
+def soft_delete(modeladmin, request, queryset):
+    """Soft delete function for selected restaurants in QuerySet list"""
+    for obj in queryset:
+        obj.delete()
+soft_delete.short_description = "Delete selected items"
+
+class PageAdmin(admin.ModelAdmin):
+
+    """Overriding of default displaying settings in restaurant's list"""
```
Comment (ivanna-ostrovets) [inline administrator/admin.py]: Please, fix this according to PEP 257.

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".
[PEP 257](https://www.python.org/dev/peps/pep-0257/#one-line-docstrings)

File: restaurant/models/restaurant.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+from django.core.exceptions import ValidationError
+from django.core.validators import MinValueValidator
+
+from django.db import models
+
+
+class RestaurantType(models.Model):
+
+    """This class is using for creating table with types of Restaurants"""
```
Comment (ivanna-ostrovets) [inline restaurant/models/restaurant.py]: Please, fix this according to PEP 257.

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".
[PEP 257](https://www.python.org/dev/peps/pep-0257/#one-line-docstrings)

File: restaurant/models/restaurant.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    rest_type = models.CharField(max_length=256, blank=False)
+    is_deleted = models.BooleanField(default=False)
+
+    class Meta(object):
+
+        """Correct displaying of restaurant type in restaurant's list"""
+
+        verbose_name=u"Restaurant type"
```
Comment (ivanna-ostrovets) [inline restaurant/models/restaurant.py]: Please, fix this according to PEP 8.

> Always surround these binary operators with a single space on either side: assignment ( = ), augmented assignment ( += , -= etc.), comparisons ( == , < , > , != , <> , <= , >= , in , not in , is , is not ), Booleans ( and , or , not ).
[PEP 8](https://www.python.org/dev/peps/pep-0008/#other-recommendations)

File: restaurant/models/restaurant.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    status = models.IntegerField(choices=RESTAURANT_STATUSES, default=0)
+    tables_count = models.IntegerField(default=0, validators = [MinValueValidator(0)])
+    description = models.TextField(max_length=256)
+    owner_id = model.ForeignKey(User)
+
+    class Meta(object):
+
+        """Display correct field in restaurant's list"""
+
+        verbose_name=u"Restaurant"
```
Comment (ivanna-ostrovets) [inline restaurant/models/restaurant.py]: Please, fix this according to PEP 8.

> Always surround these binary operators with a single space on either side: assignment ( = ), augmented assignment ( += , -= etc.), comparisons ( == , < , > , != , <> , <= , >= , in , not in , is , is not ), Booleans ( and , or , not ).
[PEP 8](https://www.python.org/dev/peps/pep-0008/#other-recommendations)

File: restaurant/models/restaurant.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        verbose_name=u"Restaurant type"
+
+    rest_type = models.CharField(max_length=256, blank=False)
+
+    def __unicode__(self):
+        return u"%s" % (self.rest_type)
+
+class Restaurant(models.Model):
+
+    """This class describe model of restaurant object,
```
Comment (ivanna-ostrovets) [inline restaurant/models/restaurant.py]: Please, fix this according to PEP 257.

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".
[PEP 257](https://www.python.org/dev/peps/pep-0257/#one-line-docstrings)

File: restaurant/models/restaurant.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        verbose_name=u"Restaurant"
+
+    name = models.CharField(max_length=256, blank=False)
+
+    def __unicode__(self):
+        """Display custom labels in restaurant list"""
+        return u"%s %s" % (self.type_id, self.name)
+
+    def delete(self, *args, **kwargs):
+        """Function for restaurant soft-deleting"""
```
Comment (ivanna-ostrovets) [inline restaurant/models/restaurant.py]: Please, fix this according to PEP 257.

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".
[PEP 257](https://www.python.org/dev/peps/pep-0257/#one-line-docstrings)

File: restaurant/models/restaurant.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+class RestaurantType(models.Model):
+
+    """This class is using for creating table with types of Restaurants"""
+
+    rest_type = models.CharField(max_length=256, blank=False)
+    is_deleted = models.BooleanField(default=False)
+
+    class Meta(object):
+
+        """Correct displaying of restaurant type in restaurant's list"""
```
Comment (ivanna-ostrovets) [inline restaurant/models/restaurant.py]: Please, fix this according to PEP 257.

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".
[PEP 257](https://www.python.org/dev/peps/pep-0257/#one-line-docstrings)

File: restaurant/models/restaurant.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+    """This class describe model of restaurant object,
+        that creates from admin panel.
+    """
+
+    ACTIVE = 0
+    DELETED = 1
+    HIDDEN = 2
+
+    RESTAURANT_STATUSES =(
```
Comment (ivanna-ostrovets) [inline restaurant/models/restaurant.py]: Please, fix this according to PEP 8.

> Always surround these binary operators with a single space on either side: assignment ( = ), augmented assignment ( += , -= etc.), comparisons ( == , < , > , != , <> , <= , >= , in , not in , is , is not ), Booleans ( and , or , not ).
[PEP 8](https://www.python.org/dev/peps/pep-0008/#other-recommendations)

File: restaurant/models/restaurant.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        (DELETED,'deleted'),
+        (HIDDEN,'hidden'),
+    )
+
+    name = models.CharField(max_length=256, blank=False)
+    logo = models.CharField(max_length=256, default="Logo_added")
+    location = models.CharField(max_length=256, blank=False)
+    type_id = models.ForeignKey(RestaurantType, blank=True, null=True)
+    status = models.IntegerField(choices=RESTAURANT_STATUSES, default=0)
+    tables_count = models.IntegerField(default=0, validators = [MinValueValidator(0)])
```
Comment (ivanna-ostrovets) [inline restaurant/models/restaurant.py]: Please, fix this according to PEP 8.

>Don't use spaces around the = sign when used to indicate a keyword argument or a default parameter value.
[PEP 8](https://www.python.org/dev/peps/pep-0008/#other-recommendations)

File: administrator/admin.py [hunk 10]
```diff
@@ -3,4 +3,26 @@
 
 from django.contrib import admin
 
-# Register your models here.
+from restaurant.models import Restaurant
+
+
+def soft_delete(modeladmin, request, queryset):
+    """Soft delete function for selected restaurants in QuerySet list"""
```
Comment (ivanna-ostrovets) [inline administrator/admin.py]: Please, fix this according to PEP 257.

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".
[PEP 257](https://www.python.org/dev/peps/pep-0257/#one-line-docstrings)

File: restaurant/models/restaurant.py [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+    rest_type = models.CharField(max_length=256, blank=False)
+
+    def __unicode__(self):
+        return u"%s" % (self.rest_type)
+
+class Restaurant(models.Model):
+
+    """This class describe model of restaurant object,
+        that creates from admin panel.
```
Comment (ivanna-ostrovets) [inline restaurant/models/restaurant.py]: Please, indent this with 4 spaces.

File: restaurant/models/restaurant.py [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    """
+
+    ACTIVE = 0
+    DELETED = 1
+    HIDDEN = 2
+
+    RESTAURANT_STATUSES =(
+        (ACTIVE,'active'),
```
Comment (ivanna-ostrovets) [inline restaurant/models/restaurant.py]: Please, add whitespace after comma according to PEP 8.

File: restaurant/models/restaurant.py [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+        (DELETED,'deleted'),
```
Comment (ivanna-ostrovets) [inline restaurant/models/restaurant.py]: Please, add whitespace after comma according to PEP 8.

File: restaurant/models/restaurant.py [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+        (HIDDEN,'hidden'),
```
Comment (ivanna-ostrovets) [inline restaurant/models/restaurant.py]: Please, add whitespace after comma according to PEP 8.

**Review by ivanna-ostrovets — CHANGES_REQUESTED** [review]

**Pull request merged**